Application/src/Model/diagramObject.java

```diff
@@ -0,0 +1,218 @@
+package Model;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+/**
+ * @author Rashad Kamsheh & Isabelle Törnqvist
+ * @since 2017-10-16
+ */
+
+public class diagramObject {
```

Comment:
Multiple public classes in one file?
How is your IDE not crying and complaining?
this should not be allowed.

Reply:
apparently it is allowed in this case because all the other public classes are nested within the diagramObject class.

Reply:
Something doesnt have to be good even though it's allowed